lib/Model/ModelManager.php

```diff
@@ -554,16 +540,24 @@ public function fixInstance(string $instance): string {
 	 * @return string
 	 */
 	public function generateLinkToCircle(string $singleId): string {
-		$path = $this->configService->getAppValue(ConfigService::ROUTE_TO_CIRCLE);
+		if (!$this->pathLinkGenerated) {
+			$this->pathLinkGenerated = true;
+			$path = $this->configService->getAppValue(ConfigService::ROUTE_TO_CIRCLE);
+			$pos = strpos($path, '.');
+			if (!$pos) {
+				return '';
+			}
 
-		try {
-			if ($path !== '') {
-				return $this->urlGenerator->linkToRoute($path, ['singleId' => $singleId]);
+			if (!$this->appManager->isInstalled(substr($path, 0, $pos))) {
```

Comment:
This should check if the app is enabled, since routes are only loaded in that case.

Reply:
isInstalled() check if app is enabled, as said in method phpdoc :)

Reply:
I'd say even better reason to use the other one to be more explicit 🙈 We should actually deprecate that app manager method then.